### PR TITLE
samples: portability: Resolve control flow issue

### PR DESCRIPTION
**Pull request by spoorthik**
Resolve a minor control flow issue in CMSIS RTOS v1 and CMSIS RTOS v2
sample.

Fixes: #12295

Signed-off-by: Spoorthi K <spoorthi.k@intel.com>

### CODE REVIEW

File: samples/portability/cmsis_rtos_v1/timer_synchronization/src/main.c [hunk 1]
```diff
@@ -41,16 +41,18 @@ void read_msg_callback(void const *arg)
...
-void send_msg_thread(void)
+int send_msg_thread(void)
 {
 	osStatus status;
 
 	status = osMessagePut(message_id, data, osWaitForever);
 	if (status != osOK) {
 		printk("\n**Error sending message to message queue**\n");
+		return 1;
 	} else {
```
Comment (rgundi) [inline samples/portability/cmsis_rtos_v1/timer_synchronization/src/main.c]: you can take out the "else" here as the "if" statement returns.

**Review by rgundi — CHANGES_REQUESTED** [review]
You need to split this up into 2 commits, one each for v1 and v2. Also, the title of the commit should be something like cmsis_rtos_v1: samples: Resolve control flow issue.

**Comment by spoorthik** [comment]
@rgundi: I have addressed your review comments. Please have a look. Thanks.

File: samples/portability/cmsis_rtos_v2/timer_synchronization/src/main.c [hunk 2]
```diff
@@ -59,7 +59,9 @@ void send_msg_thread(void)
 		printk("Wrote to message queue: %d\n", data);
```
Comment (rgundi) [inline samples/portability/cmsis_rtos_v2/timer_synchronization/src/main.c]: Move the return 0 here and then take out the "else".

**Review by rgundi — CHANGES_REQUESTED** [review]

File: samples/portability/cmsis_rtos_v2/timer_synchronization/src/main.c (hunk 2 above)
Comment (spoorthik) [reply samples/portability/cmsis_rtos_v2/timer_synchronization/src/main.c]: @rgundi: Done, Thanks

Comment (spoorthik) [reply samples/portability/cmsis_rtos_v2/timer_synchronization/src/main.c]: @rgundi: Done, Thanks

**Review by rgundi — APPROVED** [review]
LGTM

**Comment by spoorthik** [comment]
@nashif : Would you please review?

**Pull request merged**